default/python/AI/fleet_orders.py

```diff
@@ -264,29 +285,25 @@ def is_valid(self):
 
     def issue_order(self):
         if not super(OrderResupply, self).issue_order():
-            return
+            return False
         fleet_id = self.fleet.id
         system_id = self.target.get_system().id
         fleet = self.fleet.get_object()
-        if system_id not in [fleet.systemID, fleet.nextSystemID]:
-            # if self.order_type == AIFleetOrderType.ORDER_MOVE:
-            #     dest_id = system_id
-            # else:
-            # if self.order_type == AIFleetOrderType.ORDER_REPAIR:
-            #     fo.issueAggressionOrder(fleet_id, False)
+        if system_id == fleet.systemID:
+            if foAI.foAIstate.get_fleet_role(fleet_id) == MissionType.EXPLORATION:
```

Comment:
Can you give an example of role when target system  will not be explored?

Reply:
Although any fleet type being in (or merely passing through) the system is sufficient to toggle the 'explored' flag, the 'needsEmergencyExploration' list is reserved for more particular scouting needs, generally related to unexpected visibility trouble.  Since a common source of unexpected visibility trouble is reduced visibility range, which is more likely to be an issue, or be a bigger issue, for a non-scouting fleet.  It's my recollection that this was especially common back when ion storms reduced visibility range but did not affect not stealth.  It's possible that a more complex assessment could be done instead, but given as the whole 'needsEmergencyExploration' tactic is a bit more of a "Hail Mary" attempt in the face of unexpected failure rather than a precisely calculated tactic, I think the simple requirement of a scouting fleet being present is still the right rule.